src/filtering/favouriteFiltersDialogue.component.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  Grid,
+  TextField,
+} from '@mui/material';
+import React from 'react';
+import { useAddFavouriteFilter } from '../api/favouriteFilters';
+import { FavouriteFilter } from '../app.types';
+import { FilterPageHelp } from './filterDialogue.component';
+import FilterInput from './filterInput.component';
+import { Token } from './filterParser';
+
+export interface FavouriteFiltersDialogueProps {
+  open: boolean;
+  onClose: () => void;
+  channels: Token[];
+}
+
+interface FavouriteFilterTokenised {
+  name: string;
+  filter: Token[];
+}
+
+interface FavouriteFilterError {
+  name?: string;
```

Comment:
Why does the error have a name property? As far as I can see it's never used and makes the typing more complex when it could just be a string

Reply:
The name property is added for future-proofing, allowing for potential extensions like checking for duplicate names. While this hasn't been implemented in the backend yet, it could be useful as a user could create multiple filters with the same name, which doesn't look ideal in the frontend.